src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java

```diff
@@ -452,85 +565,81 @@ public boolean shouldSkipMonitoredTripCheck() throws Exception {
     }
 
     /**
-     * Calculate the next possible time that the itinerary is possible, advancing the targetZonedDateTime and setting
-     * the targetZonedDateTime, targetDate, otpResponse and matchingItinerary. Returns false if an error occurs or a
-     * matching itinerary isn't found on the next possible date the trip should be monitored on.
+     * Advance the journey state data to point to the next actively monitored date and time that the scheduled itinerary
+     * could occur. This does not make a request to the trip planner, instead it will offset the matching itinerary and
+     * journey state data the appropriate number of days and then recalculate the new scheduled time based on the
+     * updated time in the updated itinerary.
      */
-    private boolean calculateNextItinerary() {
-        // double check that this trip is in fact active and throw an error if it is not
-        if (trip.isInactive()) {
-            throw new RuntimeException("An attempt was made to find the next possible active date of an inactive monitored trip.");
-        }
-
-        // if the trip is not active on the current zoned date time, advance until a day is found when the trip is
-        // active.
+    private void advanceToNextActiveTripDate() throws URISyntaxException {
+        // Advance the target date/time until a day is found when the trip is active.
         while (!trip.isActiveOnDate(targetZonedDateTime)) {
             targetZonedDateTime = targetZonedDateTime.plusDays(1);
         }
-        targetDate = targetZonedDateTime.format(DateTimeUtils.DEFAULT_DATE_FORMATTER);
 
-        // execute trip plan request for the target time
-        LOG.info("Attempting to calculate next trip on target date {}", targetDate);
-        if (!calculateOtpResponse()) {
-            // failed to calculate, return false
-            return false;
+        LOG.info("Next itinerary happening on {}.", targetZonedDateTime);
+
+        // Update the matching itinerary with the expected scheduled times for when the next trip is
+        // expected to happen in a scheduled state.
+        long offsetMillis;
+        if (trip.isArriveBy()) {
+            // For arrive by trips, increment the matching itinerary end time as long as it does not exceed the target
+            // zoned date time.
+            //
+            // Example: The new target time is June 15 at 9am and the previous matching itinerary ended on June 13 at
+            // 8:50am. In this case, the matching itinerary will be incremented two days so the updated matching
+            // itinerary ends at 8:50am on June 15.
+            ZonedDateTime newEndTime = DateTimeUtils.makeOtpZonedDateTime(
+                new Date(matchingItinerary.getScheduledEndTimeEpochMillis())
+            );
+            do {
+                newEndTime = newEndTime.plusDays(1);
```

Comment:
Same thing as for the "leave now" trip where notifications are not sent.
To reproduce with FDOT:
- Run https://github.com/opentripplanner/otp-react-redux/pull/289
- Plan and save a trip on SunRail using an "arrive by" time today
- The log will say an itinerary happens today.
- This line will execute and the monitoring will be set for the following day.

Reply:
I believe 0a834f09c6880b1a8d3d57e4870ef7bbf43be0a8 has fixed this.